gateway-spi/src/main/java/org/apache/knox/gateway/util/URLUtils.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with this
+ * work for additional information regarding copyright ownership. The ASF
+ * licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.knox.gateway.util;
+
+import java.net.URI;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+
+public class URLUtils {
+
+  /**
+   * A method that decode url and concat query params, return result as URI.
+   *
+   * @param encoded encoded url
+   * @param queryStr query params
+   * @return decoded URI
+   */
+  public static URI getDecodeUri(String encoded, String queryStr) {
```

Comment:
nit: I think there is a typo here; it should be `getDecodedUri` (note the `d` at the end of `Decoded`)or the method name should be changed to a verb: `decodeUri`